src/Lykke.Service.KycReports.Services/Reports/KycReportingService.cs

```diff
@@ -0,0 +1,766 @@
+using Common;
+
+namespace Lykke.Service.KycReports.Services.Reports
+{
+    public class KycReportingService : IKycReportingService
+    {
+        private readonly IKycReportsRepository _reportRepository;
+        private readonly IPersonalDataService _personalDataService;
+        private readonly ILog _log;
+        private readonly IKycStatusService _kycStatusService;
+        private readonly IClientAccountClient _clientAccountService;
+        private readonly IPartnersClient _partnersService;
+
+        public KycReportingService(
+            IKycReportsRepository reportRepository,
+            ILog log,
+            IPersonalDataService personalDataService,
+            IClientAccountClient clientAccountService,
+            IPartnersClient partnersService,
+            IKycStatusService kycStatusService)
+        {
+            _reportRepository = reportRepository;
+            _personalDataService = personalDataService;
+            _log = log;
+            _kycStatusService = kycStatusService;
+            _clientAccountService = clientAccountService;
+            _partnersService = partnersService;
+        }
+
+        public async Task<string> GetKycOfficerStatsJsonAsync(DateTime dateFrom, DateTime dateTo)
+        {
+            var jsonReportRows = (await GetKycOfficerStatsDataJsonRows(dateFrom, dateTo)).ToArray();
+
+            var reportRows = jsonReportRows
+                .Select(JsonConvert.DeserializeObject<KycOfficerStatsDataReport>)
+                .OrderByDescending(row => row.ReportDay)
+                .ThenBy(row => row.KycOfficer);
+
+            jsonReportRows = reportRows.Select(JsonConvert.SerializeObject).ToArray();
+
+            return $"'[\r\n{string.Join(", \r\n", jsonReportRows)}\r\n]'";
+        }
+
+        public async Task<string> GetKycOfficersPerformanceJsonAsync(DateTime dateFrom, DateTime dateTo)
+        {
+            var jsonReportRows = (await GetKycOfficersPerformanceJsonRows(dateFrom, dateTo)).ToArray();
+
+            var reportRows = jsonReportRows
+                .Select(JsonConvert.DeserializeObject<KycOfficersPerformanceRow>)
+                .OrderByDescending(row => row.ReportDay)
+                .ThenBy(row => row.KycOfficer)
+                .ThenBy(row => row.Operation.ToString())
+                .ThenBy(row => row.ClientEmail);
+
+            jsonReportRows = reportRows.Select(JsonConvert.SerializeObject).ToArray();
+
+            return $"'[\r\n{string.Join(", \r\n", jsonReportRows)}\r\n]'";
+        }
+
+        public async Task<string> GetKycReportDailyLeadershipDataJsonAsync(DateTime dateFrom, DateTime dateTo) {
+            var jsonReportRows = await GetKycReportDailyLeadershipJsonRows(dateFrom, dateTo);
+            return $"'[\r\n{string.Join(", \r\n", jsonReportRows)}\r\n]'";
+        }
+
+        private async Task<IEnumerable<string>> GetKycReportDailyLeadershipJsonRows(DateTime startDate, DateTime endDate)
+        {
+            if (startDate > endDate)
```

Comment:
Ideally such check should be placed into Controler and return BadRequest on check failure

Reply:
I removed 2 of 3 such checks like this one from the service code because they were excessive.
But I don't think we should refactor such code because of compatibility reasons at the present time.